src/raphael_backend_flask/auth.py

```diff
@@ -1,19 +1,51 @@
-import os
+from sqlite3 import Row
 
 from flask_httpauth import HTTPBasicAuth
 from werkzeug.security import check_password_hash, generate_password_hash
 
+from raphael_backend_flask.db import execute_sql
+
 auth = HTTPBasicAuth()
-users = {
-    user: generate_password_hash(password)
-    for user, password in (
-        item.split(":") for item in os.environ["USERS"].split(",") if item
+
+class User:
+    def __init__(self, username: str, admin: int):
+        self.username: str = username
+        self.roles: list[str] = [ "admin" ] if admin else []
+
+def get_user_sql(username: str) -> Row | None:
+    res = execute_sql(f"SELECT * FROM users WHERE username = ?", (username,))
+    if res:
+        return res[0]
+    return None
+
+def create_user_sql(username: str, password: str, admin: bool) -> None:
+    hashed = generate_password_hash(password)
+    execute_sql(
+        f"INSERT INTO users (username, password_hash, admin) VALUES (?, ?, ?)",
+        (username, hashed, 1 if admin else 0,),
     )
-}
 
+def update_user_password_sql(username: str, password: str) -> None:
+    hashed = generate_password_hash(password)
+    execute_sql(f"UPDATE users SET password_hash = '{hashed}' WHERE username = ?", (username,))
```

Comment:
`hashed` is being added via string manipulation here, which we want to avoid doing unless there's an extremely good reason for it. I doubt there's much risk of [little bobby tables](https://xkcd.com/327/) here since it's a hash rather than a direct user value, but it should be fixed regardless.